providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
```

Comment:
Good catch on the `PermitUserEnvironment` setting, I didn't consider this!
If possible, I'd like keeping the check for valid keys, though. I'd propose the following regex:

```ruby
# Allow comments starting with a `#`
# Check whether a key has a valid label/ start (DSA, RSA, Ed25519, ECDSA)
/(^\w*#|ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA)/
```

Thoughts?